Dockerfile

```diff
@@ -1,48 +1,64 @@
-    ; DEBIAN_FRONTEND=noninteractive apt-get install -qq -y --upgrade ca-certificates \
-    ; DEBIAN_FRONTEND=noninteractive apt-get install -qq -y -o Dpkg::Options::="--force-confold" \
-    nodejs npm  build-essential git  gettext-base libpq-dev libgeos-dev \
-    postgresql-client-common postgresql-client-11 \
-    apache2 libapache2-mod-wsgi-py3 \
-    gettext gettext-base  \
-    curl \
-    bash \
-    vim \
-    lynx \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
-
-RUN groupadd --gid 2500 geodata \
-    && useradd --uid 2500 --gid geodata --shell /bin/bash --create-home geodata
-
-RUN mkdir -p /var/www/vhosts/${VHOST}/conf && \
-    mkdir -p /var/www/vhosts/${VHOST}/private && \
-    mkdir -p /var/www/vhosts/${VHOST}/cgi-bin && \
-    mkdir -p /var/www/vhosts/${VHOST}/htdocs && \
-    mkdir -p /var/www/vhosts/${VHOST}/logs
-
+ENV USER geodata
+ENV GROUP geodata
 
-COPY 90-chsdi3.conf    /var/www/vhosts/mf-chsdi3/conf/
-COPY 25-mf-chsdi3.conf /etc/apache2/sites-available/000-default.conf
-RUN echo "ServerName localhost" | tee /etc/apache2/conf-available/fqdn.conf && a2enconf fqdn
+# REQUIREMENTS NOTE:
+#  - gettext-base is required for envsubst in docker-entrypoint.sh
+#  - libgeos-dev is required by shapely python package
+RUN apt-get update -qq \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -qq -y --upgrade ca-certificates \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -qq -y  \
+        libgeos-dev \
+        gettext-base \
+        apache2 libapache2-mod-wsgi-py3 \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && groupadd --gid 2500 ${GROUP} \
+    && useradd --uid 2500 --gid ${GROUP} --shell /bin/sh --create-home ${USER} \
+    && mkdir -p /var/www/vhosts/${VHOST}/conf \
+    && mkdir -p /var/www/vhosts/${VHOST}/private \
+    && mkdir -p /var/www/vhosts/${VHOST}/cgi-bin \
+    && mkdir -p /var/www/vhosts/${VHOST}/htdocs \
+    && mkdir -p /var/www/vhosts/${VHOST}/logs \
+    && pip3 install -q --upgrade pip==21.2.4 setuptools --index-url ${PYPI_URL}
 
-RUN /usr/sbin/a2enmod auth_basic authz_groupfile autoindex dir env expires filter headers http2 include mpm_event negotiation proxy proxy_http proxy_http2 rewrite setenvif status wsgi alias
+COPY --chown=${USER}:${GROUP} 90-chsdi3.conf    /var/www/vhosts/${VHOST}/conf/
+RUN echo "ServerName localhost" | tee /etc/apache2/conf-available/fqdn.conf \
+    && a2enconf fqdn \
+    && a2enmod \
+        auth_basic \
+        authz_groupfile \
+        autoindex \
+        dir \
+        env \
+        expires \
+        filter \
+        headers \
+        http2 \
+        include \
+        mpm_event \
+        negotiation \
+        proxy \
+        proxy_http \
+        proxy_http2 \
+        rewrite \
+        setenvif \
+        status \
+        wsgi \
+        alias
 
-COPY . /var/www/vhosts/${VHOST}/private/chsdi
+COPY --chown=${USER}:${GROUP} . /var/www/vhosts/${VHOST}/private/chsdi
```

Comment:
just out of interest, is it planned to start the root process in this container as geodata or why do we change the ownership here and add this user? i personally would think it is a good idea to not run the container as root and change the user in the end of the dockerfile or in the entrypoint:
https://github.com/geoadmin/service-sphinxsearch/blob/develop-docker/scripts/docker-entry.sh

Reply:
To be honest I'm not sure anymore of the reason here. But I'll create a task to review the whole user and permissions in Docker. But this task can be done as post migration cleanup.